### ISSUE TITLE
NoClassDefFoundError: javax/tools/ToolProvider on openjdk 10.0.2

### ISSUE THREAD

**Issue by heuermh** [bigdatagenomics/adam#2030]
```bash
$ mvn install
...
[INFO] --- scala-maven-plugin:3.2.2:compile (scala-compile-first) @ adam-codegen-spark2_2.11 ---
[INFO] adam/adam-codegen/src/main/scala:-1: info: compiling
[INFO] Compiling 4 source files to adam/adam-codegen/target/2.11.12/classes at 1535394877980
[ERROR] error: java.lang.NoClassDefFoundError: javax/tools/ToolProvider
[INFO] 	at scala.reflect.io.JavaToolsPlatformArchive.iterator(ZipArchive.scala:301)
[INFO] 	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
[INFO] 	at scala.reflect.io.AbstractFile.foreach(AbstractFile.scala:92)
[INFO] 	at scala.tools.nsc.util.DirectoryClassPath.traverse(ClassPath.scala:277)
[INFO] 	at scala.tools.nsc.util.DirectoryClassPath.x$15$lzycompute(ClassPath.scala:299)
[INFO] 	at scala.tools.nsc.util.DirectoryClassPath.x$15(ClassPath.scala:299)
[INFO] 	at scala.tools.nsc.util.DirectoryClassPath.packages$lzycompute(ClassPath.scala:299)
[INFO] 	at scala.tools.nsc.util.DirectoryClassPath.packages(ClassPath.scala:299)
[INFO] 	at scala.tools.nsc.util.DirectoryClassPath.packages(ClassPath.scala:264)
[INFO] 	at scala.tools.nsc.util.MergedClassPath$$anonfun$packages$1.apply(ClassPath.scala:358)
[INFO] 	at scala.tools.nsc.util.MergedClassPath$$anonfun$packages$1.apply(ClassPath.scala:358)
[INFO] 	at scala.collection.Iterator$class.foreach(Iterator.scala:891)
[INFO] 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
[INFO] 	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
[INFO] 	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
[INFO] 	at scala.tools.nsc.util.MergedClassPath.packages$lzycompute(ClassPath.scala:358)
[INFO] 	at scala.tools.nsc.util.MergedClassPath.packages(ClassPath.scala:353)
[INFO] 	at scala.tools.nsc.symtab.SymbolLoaders$PackageLoader$$anonfun$doComplete$1.apply$mcV$sp(SymbolLoaders.scala:269)
[INFO] 	at scala.tools.nsc.symtab.SymbolLoaders$PackageLoader$$anonfun$doComplete$1.apply(SymbolLoaders.scala:260)
[INFO] 	at scala.tools.nsc.symtab.SymbolLoaders$PackageLoader$$anonfun$doComplete$1.apply(SymbolLoaders.scala:260)
[INFO] 	at scala.reflect.internal.SymbolTable.enteringPhase(SymbolTable.scala:235)
[INFO] 	at scala.tools.nsc.symtab.SymbolLoaders$PackageLoader.doComplete(SymbolLoaders.scala:260)
[INFO] 	at scala.tools.nsc.symtab.SymbolLoaders$SymbolLoader.complete(SymbolLoaders.scala:211)
[INFO] 	at scala.reflect.internal.Symbols$Symbol.info(Symbols.scala:1535)
[INFO] 	at scala.reflect.internal.Mirrors$RootsBase.init(Mirrors.scala:256)
[INFO] 	at scala.tools.nsc.Global.rootMirror$lzycompute(Global.scala:73)
[INFO] 	at scala.tools.nsc.Global.rootMirror(Global.scala:71)
[INFO] 	at scala.tools.nsc.Global.rootMirror(Global.scala:39)
[INFO] 	at scala.reflect.internal.Definitions$DefinitionsClass.ObjectClass$lzycompute(Definitions.scala:257)
[INFO] 	at scala.reflect.internal.Definitions$DefinitionsClass.ObjectClass(Definitions.scala:257)
[INFO] 	at scala.reflect.internal.Definitions$DefinitionsClass.init(Definitions.scala:1390)
[INFO] 	at scala.tools.nsc.Global$Run.<init>(Global.scala:1242)
[INFO] 	at scala.tools.nsc.Driver.doCompile(Driver.scala:31)
[INFO] 	at scala.tools.nsc.MainClass.doCompile(Main.scala:23)
[INFO] 	at scala.tools.nsc.Driver.process(Driver.scala:51)
[INFO] 	at scala.tools.nsc.Driver.main(Driver.scala:64)
[INFO] 	at scala.tools.nsc.Main.main(Main.scala)
[INFO] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[INFO] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[INFO] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[INFO] 	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
[INFO] 	at scala_maven_executions.MainHelper.runMain(MainHelper.java:164)
[INFO] 	at scala_maven_executions.MainWithArgsInFile.main(MainWithArgsInFile.java:26)
[INFO] java.lang.reflect.InvocationTargetException
[INFO] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[INFO] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[INFO] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[INFO] 	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
[INFO] 	at scala_maven_executions.MainHelper.runMain(MainHelper.java:164)
[INFO] 	at scala_maven_executions.MainWithArgsInFile.main(MainWithArgsInFile.java:26)
[ERROR] Caused by: java.lang.NoClassDefFoundError: javax/tools/ToolProvider
[INFO] 	at scala.reflect.io.JavaToolsPlatformArchive.iterator(ZipArchive.scala:301)
[INFO] 	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
[INFO] 	at scala.reflect.io.AbstractFile.foreach(AbstractFile.scala:92)
[INFO] 	at scala.tools.nsc.util.DirectoryClassPath.traverse(ClassPath.scala:277)
[INFO] 	at scala.tools.nsc.util.DirectoryClassPath.x$15$lzycompute(ClassPath.scala:299)
[INFO] 	at scala.tools.nsc.util.DirectoryClassPath.x$15(ClassPath.scala:299)
[INFO] 	at scala.tools.nsc.util.DirectoryClassPath.packages$lzycompute(ClassPath.scala:299)
[INFO] 	at scala.tools.nsc.util.DirectoryClassPath.packages(ClassPath.scala:299)
[INFO] 	at scala.tools.nsc.util.DirectoryClassPath.packages(ClassPath.scala:264)
[INFO] 	at scala.tools.nsc.util.MergedClassPath$$anonfun$packages$1.apply(ClassPath.scala:358)
[INFO] 	at scala.tools.nsc.util.MergedClassPath$$anonfun$packages$1.apply(ClassPath.scala:358)
[INFO] 	at scala.collection.Iterator$class.foreach(Iterator.scala:891)
[INFO] 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
[INFO] 	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
[INFO] 	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
[INFO] 	at scala.tools.nsc.util.MergedClassPath.packages$lzycompute(ClassPath.scala:358)
[INFO] 	at scala.tools.nsc.util.MergedClassPath.packages(ClassPath.scala:353)
[INFO] 	at scala.tools.nsc.symtab.SymbolLoaders$PackageLoader$$anonfun$doComplete$1.apply$mcV$sp(SymbolLoaders.scala:269)
[INFO] 	at scala.tools.nsc.symtab.SymbolLoaders$PackageLoader$$anonfun$doComplete$1.apply(SymbolLoaders.scala:260)
[INFO] 	at scala.tools.nsc.symtab.SymbolLoaders$PackageLoader$$anonfun$doComplete$1.apply(SymbolLoaders.scala:260)
[INFO] 	at scala.reflect.internal.SymbolTable.enteringPhase(SymbolTable.scala:235)
[INFO] 	at scala.tools.nsc.symtab.SymbolLoaders$PackageLoader.doComplete(SymbolLoaders.scala:260)
[INFO] 	at scala.tools.nsc.symtab.SymbolLoaders$SymbolLoader.complete(SymbolLoaders.scala:211)
[INFO] 	at scala.reflect.internal.Symbols$Symbol.info(Symbols.scala:1535)
[INFO] 	at scala.reflect.internal.Mirrors$RootsBase.init(Mirrors.scala:256)
[INFO] 	at scala.tools.nsc.Global.rootMirror$lzycompute(Global.scala:73)
[INFO] 	at scala.tools.nsc.Global.rootMirror(Global.scala:71)
[INFO] 	at scala.tools.nsc.Global.rootMirror(Global.scala:39)
[INFO] 	at scala.reflect.internal.Definitions$DefinitionsClass.ObjectClass$lzycompute(Definitions.scala:257)
[INFO] 	at scala.reflect.internal.Definitions$DefinitionsClass.ObjectClass(Definitions.scala:257)
[INFO] 	at scala.reflect.internal.Definitions$DefinitionsClass.init(Definitions.scala:1390)
[INFO] 	at scala.tools.nsc.Global$Run.<init>(Global.scala:1242)
[INFO] 	at scala.tools.nsc.Driver.doCompile(Driver.scala:31)
[INFO] 	at scala.tools.nsc.MainClass.doCompile(Main.scala:23)
[INFO] 	at scala.tools.nsc.Driver.process(Driver.scala:51)
[INFO] 	at scala.tools.nsc.Driver.main(Driver.scala:64)
[INFO] 	at scala.tools.nsc.Main.main(Main.scala)
[INFO] 	... 6 more
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] ADAM_2.11 .......................................... SUCCESS [ 11.086 s]
[INFO] ADAM_2.11: Avro-to-Dataset codegen utils ........... FAILURE [  5.907 s]
[INFO] ADAM_2.11: Core .................................... SKIPPED
[INFO] ADAM_2.11: APIs for Java, Python ................... SKIPPED
[INFO] ADAM_2.11: CLI ..................................... SKIPPED
[INFO] ADAM_2.11: Assembly ................................ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 17.202 s
[INFO] Finished at: 2018-08-27T13:34:41-05:00
[INFO] Final Memory: 21M/80M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal net.alchim31.maven:scala-maven-plugin:3.2.2:compile (scala-compile-first) on project adam-codegen-spark2_2.11: wrap: org.apache.commons.exec.ExecuteException: Process exited with an error: 240 (Exit value: 240) -> [Help 1]

$ java --version
openjdk 10.0.2 2018-07-17
OpenJDK Runtime Environment (build 10.0.2+13-Ubuntu-1ubuntu0.18.04.1)
OpenJDK 64-Bit Server VM (build 10.0.2+13-Ubuntu-1ubuntu0.18.04.1, mixed mode)

$ mvn --version
Apache Maven 3.5.2
Maven home: /usr/share/maven
Java version: 10.0.2, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-11-openjdk-amd64
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.15.0-20-generic", arch: "amd64", family: "unix"
```

**Comment by heuermh** [comment]
Closing as WontFix; need to wait on Spark supporting JDK 11.
